src/nupic/algorithms/KNNClassifier.py

```diff
@@ -845,31 +869,35 @@ def getPartitionId(self, i):
 
   def getPartitionIdPerPattern(self):
     """
-    Returns a list of numPatterns elements where the i'th position contains
-    the integer partition Id associated with pattern i. If pattern i had no
-    partition Id, it's value will be numpy.inf
+    Gets all the partition ids.
+
+    TODO: This method doesn't make sense to me. --Matt
+
+    :returns: a list of numPatterns elements where the i'th position contains
+        the integer partition Id associated with pattern i. If pattern i had no
+        partition Id, it's value will be numpy.inf
```

Comment:
It seems like a useful accessor unless there is some other way to get `_partitionIdList`. Just because it isn't used in NuPIC doesn't mean it isn't useful for clients.